File: tests/Generator/resource/go/complex/reference_property_type.go

// Represents a reference to a definition type
type ReferencePropertyType struct {
    Deprecated bool `json:"deprecated"`
    Description string `json:"description"`
    Nullable bool `json:"nullable"`
    Type string `json:"type"`
    Target string `json:"target"`
    Template map[string]string `json:"template"`
}

File: tests/Generator/resource/rust/complex/reference_property_type.rs
use serde::{Serialize, Deserialize};
use property_type::PropertyType;

// Represents a reference to a definition type
#[derive(Serialize, Deserialize)]
pub struct ReferencePropertyType {
    #[serde(rename = "deprecated")]
    deprecated: Option<bool>,

    #[serde(rename = "description")]
    description: Option<String>,

    #[serde(rename = "nullable")]
    nullable: Option<bool>,

    #[serde(rename = "type")]
    _type: Option<String>,

    #[serde(rename = "target")]
    target: Option<String>,

    #[serde(rename = "template")]
    template: Option<std::collections::HashMap<String, String>>,

}

